libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -685,33 +695,36 @@ static CK_RV prvRsaKeyAttParse( const CK_ATTRIBUTE * pxAttribute,
     CK_RV xResult = CKR_OK;
     int32_t lMbedTLSResult = 0;
     CK_BBOOL xBool;
-    mbedtls_rsa_context * pxRsaContext = ( mbedtls_rsa_context * ) pxMbedContext->pk_ctx;
 
-    configASSERT( pxRsaContext != NULL );
+    configASSERT( ( mbedtls_rsa_context * ) pxMbedContext->pk_ctx != NULL );
+    mbedtls_rsa_context * pxRsaContext = ( mbedtls_rsa_context * ) pxMbedContext->pk_ctx;
 
     switch( pxAttribute->type )
     {
         case ( CKA_CLASS ):
         case ( CKA_KEY_TYPE ):
         case ( CKA_LABEL ):
+            LogDebug( ( "Received attribute type %0X and ignored it.", pxAttribute->type ) );
             /* Do nothing. These values were parsed previously. */
             break;
 
         case ( CKA_SIGN ):
         case ( CKA_TOKEN ):
+            LogDebug( ( "Found CKA_TOKEN or CKA_SIGN." ) );
             ( void ) memcpy( &xBool, pxAttribute->pValue, pxAttribute->ulValueLen );
 
             /* See explanation in prvCheckValidSessionAndModule for this exception. */
             /* coverity[misra_c_2012_rule_10_5_violation] */
             if( xBool != ( CK_BBOOL ) CK_TRUE )
             {
-                LogError( ( "Only RSA private keys with signing permissions supported." ) );
+                LogError( ( "Failed to parse RSA private key template. Expected signing permissions to be supported or for the object to be a token type." ) );
```

Comment:
The second statement is not clear...Can it be changed to something like "Either token type object OR private keys with signing permissions are supported"?